sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -421,7 +421,13 @@ trait V2TableWriteExec extends V2CommandExec with UnaryExecNode {
 
 trait WritingSparkTask[W <: DataWriter[InternalRow]] extends Logging with Serializable {
 
-  protected def write(writer: W, row: InternalRow): Unit
+  protected def write(writer: W, row: InternalRow): Unit = {}
```

Comment:
do we still need this method?

Reply:
There is a DeltaWriter that has custom implementations of `write`:
https://github.com/apache/spark/blob/0b8e0998c67331bc82428d807476584f3d21f934/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala#L529-L559

Reply:
We can migrate that to `writeAll`, but I am not sure if we can remove this `write` method in the trait `WritingSparkTask`.

Reply:
isn't it internal API?

Reply:
True. Let me update this.